packages/with_bloc/lib/src/with_bloc.dart

```diff
@@ -48,8 +56,45 @@ class WithBloc<BlocType extends ValueNotifier<StateType>, StateType>
 
   /// The parameters the BLoC depends upon.
   ///
+  /// If [onInputsChange] is provided, the [onInputsChange] callback will be called when these inputs change.
+  ///
+  /// Otherwise we will create a new BLoC when the [inputs] change and dispose the old BLoC.
   /// If these change, the BLoC will be recreated and the old BloC will be disposed.
-  final List<dynamic> inputs;
+  final List inputs;
+
+  /// This callback will be called when ever the [inputs] change.
+  /// This will receive the current [BlocType], the previous [inputs] and the new [inputs].
+  ///
+  /// Can be `null`.
+  ///
+  /// Use this callback to update (call some method) instead of the default behaviour of recreating the BLoC.
+  ///
+  /// The return value should be a [bool], this will decide if we recreate the BLoC based on the inputs change.
+  /// Return `true` for recreating the BLoC and `false` for not recreating it.
+  ///
+  /// This allows you to handle different input changes differently, e.g.:
+  /// {@tool sample}
+  ///
+  /// ```dart
+  /// WithBloc(
+  ///   createBloc: () => Bloc(),
+  ///   inputs: [locale, isVip],
+  ///   onInputsChange: (bloc, {previousInputs, newInputs}) {
```

Comment:
@HenriBeck I think I would have inverted this. Return `true ` to signal that the callback has done it's job successfully.

Reply:
Yeah, sounds also good. Then it's also in sync with how the `Notifications` in Flutter work, e.g. returning true to cancel the bubbling